airbyte-cdk/python/airbyte_cdk/sources/lcc/requesters/paginators/interpolated_paginator.py

```diff
@@ -0,0 +1,26 @@
+#
+# Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Any, List, Mapping, Optional
+
+import requests
+from airbyte_cdk.sources.lcc.interpolation.interpolated_mapping import InterpolatedMapping
+from airbyte_cdk.sources.lcc.interpolation.jinja import JinjaInterpolation
+from airbyte_cdk.sources.lcc.requesters.paginators.paginator import Paginator
+
+
+class InterpolatedPaginator(Paginator):
+    def __init__(self, next_page_token: Mapping[str, str], config):
+        self._interpolation = InterpolatedMapping(next_page_token, JinjaInterpolation())
+        self._next_page_token = next_page_token
+        self._config = config
+
+    def next_page_token(self, response: requests.Response, last_records: List[Mapping[str, Any]]) -> Optional[Mapping[str, Any]]:
+        decoded_response = response.json()
```

Comment:
just something for the future: is it be valid to assume the response will always be json? if we have a "decoder" component to decode JSON/XML/grpc --> `Mapping` it would go here right?

Reply:
Good call. I'll pass in a decoder to future-proof this